src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -1685,6 +1686,32 @@ public Consumer<Message<String>> consumeSaStopped() {
         };
     }
 
+    @Bean
+    @Transactional
+    public Consumer<Message<String>> consumeSaFailed() {
+        return message -> {
+            String receiver = message.getHeaders().get(HEADER_RECEIVER, String.class);
```

Comment:
Quite cosmetic, but if feel uncomfortable with naming receiver the identifier of the 'real' receiver object you then get/read as receiverObj. I'd go the other way, with "String receiverID" and "Receiver receiver".